tests/v2/validation/charts/webhook_test.go

```diff
@@ -0,0 +1,139 @@
+package charts
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/charts"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeconfig"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type WebhookTestSuite struct {
+	suite.Suite
+	client       *rancher.Client
+	session      *session.Session
+	clusterList  string
+	chartVersion string
+}
+
+func (w *WebhookTestSuite) TearDownSuite() {
+	w.session.Cleanup()
+}
+
+func (w *WebhookTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	w.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+
+	w.client = client
+
+	// Get clusterName from config yaml
+	w.clusterList = client.RancherConfig.ClusterName
+	w.chartVersion, err = client.Catalog.GetLatestChartVersion(charts.RancherWebhookName)
+	require.NoError(w.T(), err)
+
+}
+
+func (w *WebhookTestSuite) ValidateEscalationCheck(client *rancher.Client) {
+
+	getAdminRole, err := client.Management.GlobalRole.ByID(admin)
+	require.NoError(w.T(), err)
+	updatedAdminRole := *getAdminRole
+	updatedAdminRole.NewUserDefault = true
+
+	_, err = client.Management.GlobalRole.Update(getAdminRole, updatedAdminRole)
+	require.Error(w.T(), err)
+	errMessage := "admission webhook \"rancher.cattle.io.globalroles.management.cattle.io\" denied the request"
+
+	assert.Contains(w.T(), err.Error(), errMessage)
+}
+
+func (w *WebhookTestSuite) ValidatePodLogs(podName, clusterID string) {
+
+	podLogs, err := kubeconfig.GetPodLogs(w.client, clusterID, podName, charts.RancherWebhookNamespace)
+	require.NoError(w.T(), err)
+	delimiter := "\n"
+	segments := strings.Split(podLogs, delimiter)
+
+	for _, segment := range segments {
```

Comment:
Should this fail for _any_ `error` in the webhook pods?

Reply:
yes, as webhook logs should not contain errors, for any errors this test would fail and print the failed log.